examples.py/Library/PDF Export/Complex3D.pyde

```diff
@@ -0,0 +1,175 @@
+"""
+PDF Complex by Marius Watz (workshop.evolutionzone.com).
+
+Example using PDF to output complex 3D geometry for print.
+Press "s" to save a PDF.
+"""
+
+add_library('pdf')  # import processing.pdf.*
+
+# Trig lookup tables (sinLUT and cosLUT) borrowed from Toxi.
+# Cryptic but effective
+SINCOS_PRECISION = 1.0
+SINCOS_LENGTH = int(360.0 / SINCOS_PRECISION)
+
+# System data
+do_save = False
+
+def setup():
+    global sinLUT, cosLUT, num, pt, style
+    size(600, 600, OPENGL)
+    frameRate(24)
+    background(255)
+
+    # Fill the tables
+    sinLUT = [sin(i * DEG_TO_RAD * SINCOS_PRECISION)
+              for i in range(SINCOS_LENGTH)]
+    cosLUT = [cos(i * DEG_TO_RAD * SINCOS_PRECISION)
+              for i in range(SINCOS_LENGTH)]
+    num = 150
+    pt = [0.0 for _ in range(6 * num)]  # rotx, roty, deg, rad, w, speed
+    style = [0 for _ in range(2 * num)]  # color, render style
+
+    # Set up arc shapes
+    index = 0
+    for i in range(num):
+        pt[index] = random(TWO_PI)  # Random X axis rotation
+        index += 1
+        pt[index] = random(TWO_PI)  # Random Y axis rotation
+        index += 1
+        
+        pt[index] = random(60, 80)  # Short to quarter-circle arcs
+        if random(100) > 90: pt[index] = int(random(8, 27) * 10.0)
+        index += 1  # correction added on March 19th 2017 by Marius Watz
```

Comment:
Talked to him on Twitter... https://twitter.com/mariuswatz/status/843518842890588160
I've changed the code a bit more and removed this comment... but it's still true.